Update validator well known keys

hewison-chris opened the pull request:
This is in preparation for issue 1248 which will enforce sorted inputs
and outputs within transactions. To keep the order of the freeze outputs
the same as the nodes used in unit tests the keys had to be updated to
have the same prefix so that the public key data was ascending.
The node keys now have bech32 addresses starting with `boa1xzval` or `boa1xrval`.
Two patterns are needed as the conversion to bech32 does not exactly follow alphanumeric sorting.
e.g. boa1xzval2a3cdxv28n6slr62wlczslk3juvk7cu05qt3z55ty2rlfqfc6egsh2

hewison-chris committed:
Update validator well known keys

This is in preparation for issue 1248 which will enforce sorted inputs
and outputs within transactions. To keep the order of the freeze outputs
the same as the nodes used in unit tests the keys had to be updated to
have the same prefix so that the public key data was ascending.
The node keys now have bech32 addresses starting with `boa1xzval`.
e.g. boa1xzval2a3cdxv28n6slr62wlczslk3juvk7cu05qt3z55ty2rlfqfc6egsh2

linked0 reviewed (approved):
LGTM. There are CI errors that are not related to this PR though.

Geod24 commented:
Can you rebase ?

Geod24 commented:
Rebased

Geod24 commented:
```
Unittesting agora.test.Flash..
================================ ASSERT HANDLER ===============================
[source/agora/test/Flash.d:191] Assertion thrown during test: Agora node not initialized
/Users/runner/work/agora/agora/source/agora/test/Flash.d:191 agora.api.FullNode.API agora.test.Flash.TestFlashNode.getAgoraClient(immutable(char)[], core.time.Duration) [0x103dc57e5]
/Users/runner/work/agora/agora/source/agora/flash/Node.d:1366 agora.flash.Node.ThinFlashNode agora.flash.Node.ThinFlashNode.__ctor(agora.flash.Config.FlashConfig, immutable(char)[], geod24.bitblob.BitBlob!(64uL).BitBlob, agora.script.Engine.Engine, agora.common.Task.ITaskManager, immutable(char)[]) [0x103dc5270]
/Users/runner/work/agora/agora/source/agora/test/Flash.d:149 agora.test.Flash.TestFlashNode agora.test.Flash.TestFlashNode.__ctor(agora.flash.Config.FlashConfig, shared(geod24.Registry.Registry!(agora.test.Base.TestAPI).Registry)*, immutable(char)[], agora.test.Flash.DatabaseStorage, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashAPI).Registry)*, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashListenerAPI).Registry)*) [0x103dc51a5]
/Users/runner/work/agora/agora/submodules/localrest/source/geod24/LocalRest.d:739 void geod24.LocalRest.RemoteAPI!(agora.test.Flash.TestFlashAPI, agora.test.Base.Serializer).RemoteAPI.spawned!(agora.test.Flash.TestFlashNode).spawned(geod24.concurrency.Channel!(geod24.LocalRest.Connection).Channel, immutable(char)[], int, agora.flash.Config.FlashConfig, shared(geod24.Registry.Registry!(agora.test.Base.TestAPI).Registry)*, immutable(char)[], agora.test.Flash.DatabaseStorage, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashAPI).Registry)*, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashListenerAPI).Registry)*).runNode() [0x103e86e40]
/Users/runner/work/agora/agora/submodules/localrest/source/geod24/LocalRest.d:845 nothrow void geod24.LocalRest.RemoteAPI!(agora.test.Flash.TestFlashAPI, agora.test.Base.Serializer).RemoteAPI.spawned!(agora.test.Flash.TestFlashNode).spawned(geod24.concurrency.Channel!(geod24.LocalRest.Connection).Channel, immutable(char)[], int, agora.flash.Config.FlashConfig, shared(geod24.Registry.Registry!(agora.test.Base.TestAPI).Registry)*, immutable(char)[], agora.test.Flash.DatabaseStorage, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashAPI).Registry)*, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashListenerAPI).Registry)*) [0x103e86adb]
/Users/runner/work/agora/agora/submodules/localrest/source/geod24/LocalRest.d:631 void geod24.LocalRest.RemoteAPI!(agora.test.Flash.TestFlashAPI, agora.test.Base.Serializer).RemoteAPI.spawn!(agora.test.Flash.TestFlashNode).spawn(agora.flash.Config.FlashConfig, shared(geod24.Registry.Registry!(agora.test.Base.TestAPI).Registry)*, immutable(char)[], agora.test.Flash.DatabaseStorage, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashAPI).Registry)*, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashListenerAPI).Registry)*, core.time.Duration, immutable(char)[], int).__lambda12() [0x103e8685f]
context.d:46 void core.thread.context.Callable.opCall() [0x104e917ce]
threadbase.d:429 void core.thread.threadbase.ThreadBase.run() [0x104e9532d]
osthread.d:349 void core.thread.osthread.Thread.run() [0x104e92884]
osthread.d:2351 thread_entryPoint [0x104e92e68]
??:? _pthread_start [0x7fff6985a108]
```

That one's not spurious it seems

hewison-chris committed:
Flash test - fix address for first node

Some tests were assuming that NODE2 address corresponded to the index of
 the first node in the test GenesisBlock.
 Now the tests look up the address of the first node via the
 `genesis_validator_keys` which are ordered to match the enrollments
 which are sorted by `utxo_key`.

hewison-chris commented:
> ```
> Unittesting agora.test.Flash..
> ================================ ASSERT HANDLER ===============================
> [source/agora/test/Flash.d:191] Assertion thrown during test: Agora node not initialized
> /Users/runner/work/agora/agora/source/agora/test/Flash.d:191 agora.api.FullNode.API agora.test.Flash.TestFlashNode.getAgoraClient(immutable(char)[], core.time.Duration) [0x103dc57e5]
> /Users/runner/work/agora/agora/source/agora/flash/Node.d:1366 agora.flash.Node.ThinFlashNode agora.flash.Node.ThinFlashNode.__ctor(agora.flash.Config.FlashConfig, immutable(char)[], geod24.bitblob.BitBlob!(64uL).BitBlob, agora.script.Engine.Engine, agora.common.Task.ITaskManager, immutable(char)[]) [0x103dc5270]
> /Users/runner/work/agora/agora/source/agora/test/Flash.d:149 agora.test.Flash.TestFlashNode agora.test.Flash.TestFlashNode.__ctor(agora.flash.Config.FlashConfig, shared(geod24.Registry.Registry!(agora.test.Base.TestAPI).Registry)*, immutable(char)[], agora.test.Flash.DatabaseStorage, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashAPI).Registry)*, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashListenerAPI).Registry)*) [0x103dc51a5]
> /Users/runner/work/agora/agora/submodules/localrest/source/geod24/LocalRest.d:739 void geod24.LocalRest.RemoteAPI!(agora.test.Flash.TestFlashAPI, agora.test.Base.Serializer).RemoteAPI.spawned!(agora.test.Flash.TestFlashNode).spawned(geod24.concurrency.Channel!(geod24.LocalRest.Connection).Channel, immutable(char)[], int, agora.flash.Config.FlashConfig, shared(geod24.Registry.Registry!(agora.test.Base.TestAPI).Registry)*, immutable(char)[], agora.test.Flash.DatabaseStorage, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashAPI).Registry)*, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashListenerAPI).Registry)*).runNode() [0x103e86e40]
> /Users/runner/work/agora/agora/submodules/localrest/source/geod24/LocalRest.d:845 nothrow void geod24.LocalRest.RemoteAPI!(agora.test.Flash.TestFlashAPI, agora.test.Base.Serializer).RemoteAPI.spawned!(agora.test.Flash.TestFlashNode).spawned(geod24.concurrency.Channel!(geod24.LocalRest.Connection).Channel, immutable(char)[], int, agora.flash.Config.FlashConfig, shared(geod24.Registry.Registry!(agora.test.Base.TestAPI).Registry)*, immutable(char)[], agora.test.Flash.DatabaseStorage, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashAPI).Registry)*, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashListenerAPI).Registry)*) [0x103e86adb]
> /Users/runner/work/agora/agora/submodules/localrest/source/geod24/LocalRest.d:631 void geod24.LocalRest.RemoteAPI!(agora.test.Flash.TestFlashAPI, agora.test.Base.Serializer).RemoteAPI.spawn!(agora.test.Flash.TestFlashNode).spawn(agora.flash.Config.FlashConfig, shared(geod24.Registry.Registry!(agora.test.Base.TestAPI).Registry)*, immutable(char)[], agora.test.Flash.DatabaseStorage, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashAPI).Registry)*, shared(geod24.Registry.Registry!(agora.test.Flash.TestFlashListenerAPI).Registry)*, core.time.Duration, immutable(char)[], int).__lambda12() [0x103e8685f]
> context.d:46 void core.thread.context.Callable.opCall() [0x104e917ce]
> threadbase.d:429 void core.thread.threadbase.ThreadBase.run() [0x104e9532d]
> osthread.d:349 void core.thread.osthread.Thread.run() [0x104e92884]
> osthread.d:2351 thread_entryPoint [0x104e92e68]
> ??:? _pthread_start [0x7fff6985a108]
> ```
> 
> That one's not spurious it seems

NODE2 address is no longer first in the Test GenesisBlock. I added a commit to fix the address used in the tests.
